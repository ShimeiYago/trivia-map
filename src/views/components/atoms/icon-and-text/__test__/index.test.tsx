import { shallow, ShallowWrapper } from 'enzyme';
import { IconAndText, Props } from '..';
import MapIcon from '@mui/icons-material/Map';

let wrapper: ShallowWrapper<Props>;

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<IconAndText iconComponent={<MapIcon />} text="地図" iconPosition="left" />);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('right icon', () => {
    wrapper.setProps({
      iconPosition: 'right',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('left align', () => {
    wrapper.setProps({
      align: 'left',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('with link', () => {
    wrapper.setProps({
      link: './',
    });
    expect(wrapper).toMatchSnapshot();
  });
});
