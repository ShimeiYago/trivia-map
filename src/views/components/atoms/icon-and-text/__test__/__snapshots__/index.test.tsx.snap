// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<ForwardRef(Typography)
  component="div"
  sx={
    Object {
      "alignItems": "center",
      "columnGap": undefined,
      "display": "flex",
      "flexWrap": "wrap",
      "fontSize": undefined,
      "justifyContent": "center",
    }
  }
>
  <Memo(ForwardRef(MapIcon)) />
  地図
</ForwardRef(Typography)>
`;

exports[`Shallow Snapshot Tests left align 1`] = `
<ForwardRef(Typography)
  component="div"
  sx={
    Object {
      "alignItems": "center",
      "columnGap": undefined,
      "display": "flex",
      "flexWrap": "wrap",
      "fontSize": undefined,
      "justifyContent": "left",
    }
  }
>
  <Memo(ForwardRef(MapIcon)) />
  地図
</ForwardRef(Typography)>
`;

exports[`Shallow Snapshot Tests right icon 1`] = `
<ForwardRef(Typography)
  component="div"
  sx={
    Object {
      "alignItems": "center",
      "columnGap": undefined,
      "display": "flex",
      "flexWrap": "wrap",
      "fontSize": undefined,
      "justifyContent": "center",
    }
  }
>
  地図
  <Memo(ForwardRef(MapIcon)) />
</ForwardRef(Typography)>
`;

exports[`Shallow Snapshot Tests with link 1`] = `
<ForwardRef(Typography)
  component="div"
  sx={
    Object {
      "alignItems": "normal",
      "columnGap": undefined,
      "display": "flex",
      "flexWrap": "wrap",
      "fontSize": undefined,
      "justifyContent": "center",
    }
  }
>
  <NonStyleLink
    to="./"
  >
    <Memo(ForwardRef(MapIcon)) />
  </NonStyleLink>
  <NonStyleLink
    to="./"
  >
    地図
  </NonStyleLink>
</ForwardRef(Typography)>
`;

exports[`Shallow Snapshot Tests with onClick 1`] = `
<ForwardRef(Typography)
  component="div"
  sx={
    Object {
      "alignItems": "center",
      "columnGap": undefined,
      "display": "flex",
      "flexWrap": "wrap",
      "fontSize": undefined,
      "justifyContent": "center",
    }
  }
>
  <span
    onClick={[Function]}
  >
    <Memo(ForwardRef(MapIcon)) />
  </span>
  <span
    onClick={[Function]}
  >
    地図
  </span>
</ForwardRef(Typography)>
`;
